corehq/apps/zapier/tests/test_polling.py

```diff
@@ -0,0 +1,138 @@
+import json
+from datetime import datetime
+from unittest.mock import patch
+from django.test import SimpleTestCase, RequestFactory
+from django.contrib.auth.models import User
+
+from corehq import privileges
+from corehq.apps.api import resources
+from corehq.apps.api.resources import v0_4
+from corehq.apps.app_manager.models import Application
+from corehq.apps.api.models import ESXFormInstance
+
+from corehq.apps.zapier.api.v0_5 import ZapierApplicationResource, ZapierXFormInstanceResource
+
+
+class XFormPollingTests(SimpleTestCase):
+    def test_can_poll_with_only_zapier_privilege(self):
+        self.forms = [self._create_form(id='7', domain='test-domain')]
+        self.domain_privileges = [privileges.ZAPIER_INTEGRATION]
+        request = self._create_request('test@test.com', 'test-domain', api_key='121212')
+
+        result = self.resource.dispatch('list', request,
+            domain='test-domain', api_name='v0.5', resource_name='form')
+        rows_found = json.loads(result.content.decode('utf8'))['meta']['total_count']
+        self.assertEqual(rows_found, 1)
+
+    def setUp(self):
+        self.resource = ZapierXFormInstanceResource()
+
+        self.domain_privileges = [privileges.ZAPIER_INTEGRATION]
+        self.forms = []
+
+        resource_patcher = patch.object(self.resource, 'obj_get_list', side_effect=self._mock_get_forms)
+        resource_patcher.start()
+        self.addCleanup(resource_patcher.stop)
+
+        domain_privilege_patcher = patch.object(
+            resources, 'domain_has_privilege', side_effect=self._mock_domain_has_privilege)
+        domain_privilege_patcher.start()
+        self.addCleanup(domain_privilege_patcher.stop)
```

Comment:
FYI there's also a `@privilege_enabled('ZAPIER_INTEGRATION')` test decorator that you may find useful:
https://github.com/dimagi/commcare-hq/blob/b8b62bf3c950cedfe7692fc8cbe0710d495234e2/corehq/util/test_utils.py#L172-L173
Though since you're specifically testing that resources check privileges correctly, I'll leave it to your judgement whether that exercises appropriate code paths.